scheduler/scheduler.go

```diff
@@ -182,15 +183,23 @@ func (s *Scheduler) Sync(ctx context.Context, client schema.ClientMeta, tables s
 		}
 	}()
 	for resource := range resources {
-		vector := resource.GetValues()
-		bldr := array.NewRecordBuilder(memory.DefaultAllocator, resource.Table.ToArrowSchema())
-		scalar.AppendToRecordBuilder(bldr, vector)
-		rec := bldr.NewRecord()
-		res <- &message.SyncInsert{Record: rec}
+		select {
+		case res <- &message.SyncInsert{Record: resourceToRecord(resource)}:
+		case <-ctx.Done():
+			return nil
```

Comment:
why not return ctx.Err()?

Reply:
No good reason, I'll look at doing that 🙏